samples/dnn/siamrpnpp.py

```diff
@@ -0,0 +1,419 @@
+import argparse
+import cv2 as cv
+import glob
+import numpy as np
+import os
+
+"""
+You can download the pre-trained weights of the Tracker Model from https://drive.google.com/file/d/1PBtRDiWAIaGthMKdzyJL9qYC6ODGnXk5/view?usp=sharing
+You can download the backbone model (ResNet50 with target input) from https://drive.google.com/file/d/1jwguUwfvBa-EbDXRDfuyCrvqLPZcFZo0/view?usp=sharing
+You can download the backbone model (ResNet50 with search input) from https://drive.google.com/file/d/1L6kxi_WkdH__kfrPC_R-ZsaPD4ky52tI/view?usp=sharing
+You can download the neck model 1 (Adjusted Layers 1) from https://drive.google.com/file/d/1690vawz9b76Bwskg3CXtBb1Fd59qBsRw/view?usp=sharing
+You can download the neck model 2 (Adjusted Layers 2) from https://drive.google.com/file/d/1E7nX5MzSzVpw6a2Mm5krxp0reLrO1-je/view?usp=sharing
+You can download the head model (RPN Head) from https://drive.google.com/file/d/15Sgh1YwdH_fCnbTzhsU-HcFKpSMjPMLY/view?usp=sharing
+"""
+
+class ModelBuilder():
+    """ This class generates the SiamRPN++ Tracker Model by using Imported ONNX Nets
+    """
+    def __init__(self, backbone_search, backbone_target, neck_1, neck_2, rpn_head):
+        super(ModelBuilder, self).__init__()
+        # Build Backbone Model
+        self.backbone_search = backbone_search
+        self.backbone_target = backbone_target
+        # Build Adjusted Layer
+        self.neck_1 = neck_1
+        self.neck_2 = neck_2
+        # Build RPN_Head
+        self.rpn_head = rpn_head
+
+    def template(self, z):
+        """ Takes the template of size (1, 1, 127, 127) as an input to generate kernel
+        """
+        self.backbone_target.setInput(z)
+        outNames = ['output_1', 'output_2', 'output_3']
+        zf_1, zf_2, zf_3 = self.backbone_target.forward(outNames)
+        self.neck_1.setInput(zf_1, 'input_1')
+        self.neck_1.setInput(zf_2, 'input_2')
+        self.neck_1.setInput(zf_3, 'input_3')
+        zfs_1, zfs_2, zfs_3 = self.neck_1.forward(outNames)
+        self.zfs_1 = zfs_1
+        self.zfs_2 = zfs_2
+        self.zfs_3 = zfs_3
+
+    def track(self, x):
+        """ Takes the search of size (1, 1, 255, 255) as an input to generate classification score and bounding box regression
+        """
+        self.backbone_search.setInput(x)
+        outNames = ['output_1', 'output_2', 'output_3']
+        xf_1, xf_2, xf_3 = self.backbone_search.forward(outNames)
+        self.neck_2.setInput(xf_1, 'input_1')
+        self.neck_2.setInput(xf_2, 'input_2')
+        self.neck_2.setInput(xf_3, 'input_3')
+        xfs_1, xfs_2, xfs_3 = self.neck_2.forward(outNames)
```

Comment:
Could you try to create a single ONNX model from `backbone_search `and `neck_2` or even `backbone_search`, `neck_2` and `rpn_head`?